File: frontend/src/features/education/models/EducationResponseModel.ts
export interface EducationResponseModel {
  educationId: string;
  degree?: string;
  degreeEn: string;
  degreeFr: string;
  institution?: string;
  institutionEn: string;
  institutionFr: string;
  locationEn: string;
  locationFr: string;
  description?: string;
  descriptionEn: string;
  descriptionFr: string;
  startDate: string; // ISO date
  endDate: string | null; // ISO date or null
  period?: string;
  gpa?: string;
  type?: string;
}

File: backend/portfolio-backend/src/main/java/com/mattysleduc/portfolio/domain/educationsubdomain/presentation_layer/EducationResponseModel.java
package com.mattysleduc.portfolio.domain.educationsubdomain.presentation_layer;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class EducationResponseModel {
    private String educationId;
    private String degree;
    private String degreeEn;
    private String degreeFr;
    private String institution;
    private String institutionEn;
    private String institutionFr;
    private String locationEn;
    private String locationFr;
    private String description;
    private String descriptionEn;
    private String descriptionFr;
    private LocalDate startDate;
    private LocalDate endDate;
    private String period;
    private String gpa;
    private String type;
}
